src/App.tsx

```diff
@@ -14,11 +15,132 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGood: [],
+  };
+
+  selectHandler = (good: string) => {
+    if (!this.state.selectedGood.includes(good)) {
+      this.setState((prevState) => ({
+        selectedGood: [
+          ...prevState.selectedGood,
+          good,
+        ],
+      }));
+    }
+  };
+
+  removeHandler = (good: string) => {
+    const i = this.state.selectedGood.findIndex(item => item === good);
+
+    this.state.selectedGood.splice(i, 1);
+
+    this.setState((prevState) => ({
+      selectedGood: [
+        ...prevState.selectedGood,
+      ],
+    }));
+  };
+
+  displaySelectedGoods = () => {
+    if (this.state.selectedGood.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (this.state.selectedGood.length === 1) {
+      return `${this.state.selectedGood[0]} is selected`;
+    }
+
+    const result = [];
```

Comment:
There is no need to create this variable. Apply the join method to the selectedGoods array.